legend-engine-xt-relationalStore-spanner-execution/pom.xml

```diff
@@ -0,0 +1,100 @@
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>legend-engine</artifactId>
+        <groupId>org.finos.legend.engine</groupId>
+        <version>3.16.1-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>legend-engine-xt-relationalStore-spanner-execution</artifactId>
+    <packaging>jar</packaging>
+    <name>Legend Engine - XT - Relational Store - Spanner - Execution</name>
+
+    <dependencies>
+        <!-- ENGINE -->
+        <dependency>
+            <groupId>org.finos.legend.engine</groupId>
+            <artifactId>legend-engine-xt-relationalStore-spanner-protocol</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.finos.legend.engine</groupId>
+            <artifactId>legend-engine-xt-relationalStore-executionPlan</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.finos.legend.engine</groupId>
+            <artifactId>legend-engine-xt-relationalStore-executionPlan-connection</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.finos.legend.engine</groupId>
+            <artifactId>legend-engine-xt-relationalStore-executionPlan-connection-authentication</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.finos.legend.engine</groupId>
+            <artifactId>legend-engine-shared-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.finos.legend.engine</groupId>
+            <artifactId>legend-engine-xt-relationalStore-protocol</artifactId>
+        </dependency>
+        <!-- ENGINE -->
+
+        <!-- ECLIPSE COLLECTIONS -->
+        <dependency>
+            <groupId>org.eclipse.collections</groupId>
+            <artifactId>eclipse-collections-api</artifactId>
+        </dependency>
+        <!-- ECLIPSE COLLECTIONS -->
+
+        <!-- Guava -->
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+        </dependency>
+        <!-- Guava -->
+
+        <!-- DRIVER -->
+        <dependency>
+            <groupId>com.google.cloud</groupId>
+            <artifactId>google-cloud-spanner-jdbc</artifactId>
```

Comment:
Can we use the shaded version of the library?

Reply:
yes, we can. But it can be done in several ways that we need to discuss:
And one of the questions is should we change this dependency to please project that depends on engine.
For my point of view it isn't good thing to change project in favor of it's client - it should be in other way.
But this is a discussion to be made, let's keep that out of scope here.
  
Discussed with @vikask-gs that this change would be split int two:
1. First merge to legend-engine without enabling it in engine-server
2. Second is "enabling" spanner

Reply:
We need to make sure how opinionated we decide to be, since that could limit adoption.  Having careful on what dependencies we bring and how we bring them will impact on the experience potential clients will have while importing our libraries.  

Another angle is that this is just one component of a complex system, and we should try to limit the blast radius a new feature could have.  In the future, we might need to add YetAnotherJdbc that will conflict with the dependencies on this module.  What we do on that case?  Which feature we prioritize?  We need a clean and clear strategy on how complex features are added in ways that is nimble and less disruptive as possible.

Reply:
Downgrading guava version to reduce blast radius. Guava version would stay the same as it was prior to this PR merge